src/java/org/apache/cassandra/metrics/CassandraMetricsRegistry.java

```diff
@@ -18,85 +18,324 @@
+                                   .add(TCMMetrics.TYPE_NAME)
+                                   .add(TableMetrics.ALIAS_TYPE_NAME)
+                                   .add(TableMetrics.TYPE_NAME)
+                                   .add(TableMetrics.INDEX_TYPE_NAME)
+                                   .add(TableMetrics.INDEX_ALIAS_TYPE_NAME)
+                                   .add(ThreadPoolMetrics.TYPE_NAME)
+                                   .add(TrieMemtableMetricsView.TYPE_NAME)
+                                   .add(UnweightedCacheMetrics.TYPE_NAME)
+                                   .build();
+    }
+
     private CassandraMetricsRegistry()
     {
-        super();
     }
 
-    public Counter counter(MetricName name)
+    private static CassandraMetricsRegistry init()
     {
-        Counter counter = counter(name.getMetricName());
-        registerMBean(counter, name.getMBeanName());
+        CassandraMetricsRegistry registry = new CassandraMetricsRegistry();
+        // Adding listeners to the root registry, so that they can be notified about all metrics changes.
+        registry.addListener(registry.jmxExporter);
+        registry.addListener(registry.housekeepingListener);
+        return registry;
+    }
 
-        return counter;
+    @SuppressWarnings("rawtypes")
+    public static String getValueAsString(Metric metric)
+    {
+        if (metric instanceof Counter)
+            return Long.toString(((Counter) metric).getCount());
+        else if (metric instanceof Gauge)
+            return ((Gauge) metric).getValue().toString();
+        else if (metric instanceof Histogram)
+            return Double.toString(((Histogram) metric).getSnapshot().getMedian());
+        else if (metric instanceof Meter)
+            return Long.toString(((Meter) metric).getCount());
+        else if (metric instanceof Timer)
+            return Long.toString(((Timer) metric).getCount());
+        else
+            throw new IllegalStateException("Unknown metric type: " + metric.getClass().getName());
     }
 
-    public Counter counter(MetricName name, MetricName alias)
+    public static List<VirtualTable> createMetricsKeyspaceTables()
     {
-        Counter counter = counter(name);
-        registerAlias(name, alias);
-        return counter;
+        ImmutableList.Builder<VirtualTable> builder = ImmutableList.builder();
+        metricGroups.forEach(groupName -> {
+            // This is a very efficient way to filter metrics by group name, so make sure that metrics group name
+            // and metric type following the same order as it constructed in MetricName class.
+            final String groupPrefix = DefaultNameFactory.GROUP_NAME + '.' + groupName + '.';
+            builder.add(createSinglePartitionedKeyFiltered(VIRTUAL_METRICS,
+                                                           METRICS_GROUP_POSTFIX.apply(
+                                                               groupName),
```

Comment:
I think this would read better if it was just on the same line, same case below

Reply:
Fixed.